tickit/devices/eiger/eiger.py

```diff
@@ -0,0 +1,318 @@
+from typing import Awaitable, Callable, Optional
+
+from aiohttp import web
+
+from tickit.adapters.httpadapter import HTTPAdapter
+from tickit.adapters.interpreters.endpoints.http_endpoint import HTTPEndpoint
+from tickit.adapters.servers.http_server import HTTPServer
+from tickit.core.adapter import ConfigurableAdapter
+from tickit.core.device import ConfigurableDevice, DeviceUpdate
+from tickit.core.typedefs import SimTime
+from tickit.devices.eiger.eiger_settings import EigerSettings
+from tickit.utils.byte_format import ByteFormat
+from tickit.utils.compat.typing_compat import TypedDict
+
+from .eiger_status import EigerStatus, State
+
+DETECTOR_API = "detector/api/1.8"
+
+
+class Eiger(ConfigurableDevice):
+    """A device class for the Eiger detector.
+
+    ...
+    """
+
+    Inputs: TypedDict = TypedDict("Inputs", {"foo": bool})
+
+    Outputs: TypedDict = TypedDict("Outputs", {"bar": float})
+
+    settings: EigerSettings
+
+    status: EigerStatus
+
+    def __init__(
+        self,
+        foo: float = 0.5,
+        bar: Optional[int] = 10,
+    ) -> None:
+        """An example HTTP device constructor which configures the ... .
+
+        Args:
+            foo (bool): A flag to indicate something. Defauls to False.
+            bar (int, optional): A number to represent something. Defaults to 3600.
+        """
+        self.foo = foo
+        self.bar = bar
+        self.settings = EigerSettings()
+        self.status = EigerStatus()
+
+    async def initialize(self) -> None:
+        """Function to initialise the Eiger."""
+        self._set_state(State.IDLE)
+
+    async def arm(self) -> None:
+        """Function to arm the Eiger."""
+        self._set_state(State.READY)
+
+    async def disarm(self) -> None:
+        """Function to disarm the Eiger."""
+        self._set_state(State.IDLE)
+
+    async def trigger(self) -> str:
+        """Function to trigger the Eiger."""
+        trigger_mode = getattr(self.settings, "trigger_mode")
+        state = getattr(self.status, "state")
```

Comment:
Is there a reason we use `getattr` here? Should be able to do `state = self.status.state`

Reply:
Nah, i thought it was the better option. Do you think just setting it directly is better?

Reply:
I've changed the `getattr`s for the Eiger funcs, but for the endpoint funcs the `param` is a `str` so i can't use the same direct getting